packages/@expo/cli/src/utils/telemetry/types.ts

```diff
@@ -0,0 +1,23 @@
+import { type Actor } from '../../api/user/user';
+
+export type TelemetryEvent =
+  | 'action'
+  | 'Open Url on Device'
+  | 'Start Project'
+  | 'Serve Manifest'
+  | 'Serve Expo Updates Manifest'
+  | 'dev client start command'
+  | 'dev client run command'
+  | 'metro config'
+  | 'metro debug';
+
+export type TelemetryProperties = Record<string, any>;
+
+export type TelemetryRecord = { event: TelemetryEvent; properties?: TelemetryProperties };
+
+export interface TelemetryClient {
+  readonly isIdentified: boolean;
+  identify(actor?: Actor): Promise<void>;
+  record(event: TelemetryEvent, properties?: TelemetryProperties): void;
```

Comment:
Maybe allow `TelemetryRecord` here, that way, we can also type the properties through:

```ts
type TelemetryRecord = 
  | { event: 'reached'; }
  | { event: 'metro config'; properties: MetroConfigProperties }
  | { event: 'metro debug'; properties: { debugger: 'chrome' | 'vscode' }
  // fallback
  | { event: TelemetryEvent; properties?: TelemetryProperties }
```